src/pages/Cart/hooks/useCart.ts

```diff
@@ -0,0 +1,103 @@
+import { ResponseReturn } from '@/api';
+import { CART_PRODUCT_QUANTITY } from '@/constants';
+import { CartList, CartWithProductQuantity, ProductWithQuantity } from '@/types';
+import { ChangeEvent, useCallback, useMemo, useState } from 'react';
+import { KeyedMutator } from 'swr';
+
+export type QuantityButtonType = 'up' | 'down';
+
+const quantityCalc = (type: QuantityButtonType, quantity: number) => {
+  const { MIN, MAX } = CART_PRODUCT_QUANTITY;
+  if (type === 'up') return quantity === MAX ? MAX : quantity + 1;
+  if (type === 'down') return quantity === MIN ? MIN : quantity - 1;
+  return 0;
+};
+
+interface UseCartProps {
+  initialData: CartWithProductQuantity[];
+  mutateCart: KeyedMutator<ResponseReturn<CartList>> | undefined;
+}
+function useCart({ initialData = [] }: UseCartProps) {
+  const [cartData, setCartData] = useState<CartWithProductQuantity[]>(initialData);
+  const [checkedList, setCheckedList] = useState<number[]>(initialData.map(cart => cart.id));
```

Comment:
별도로 해당 상태를 관리하는 것 보다 { checked: boolean } 속성을 cardData에 넣어주는게 더 편하지 않을까요?
두가지 상태를 함께 관리하려니 더 복잡해지는 감이 있는 것 같아요. 

```suggestion
  const [items, setItems] = useState<CardProduct[]>((initialData) => ({....initialData, checked: true }));
```


e.g. 변경하면 좋아질 케이스
```typescript
  const totalPrice = useMemo(
    () =>
     // sum은 number 배열을 받아 모두 더해서 반환하는 함수라고 가정
      sum(items.map(item => item.checked ? item.product.price * item.product.quantity : 0)),
    [item],
  );
```

(Good to have) 상품별 체크, 삭제 등 변경이 있을때마다 서버와 동기화 시켜주는게 더 자연스럽다고 생각해요